db/migrate/20150313212337_refactor_saved_regressions.rb

```diff
@@ -0,0 +1,114 @@
+class RefactorSavedRegressions < ActiveRecord::Migration
+  def change
+    say 'Refactoring saved regressions'
+    Visualization.find_each do | v |
+      globals = JSON.parse(v.globals)
+      scatter_params = globals['Scatter']
+      timeline_params = globals['Timeline']
+      if !scatter_params.nil? and scatter_params.key? 'savedRegressions'
+        scatter_regressions = []
+        scatter_params['savedRegressions'].each do |regression|
+          keys = regression.keys
+          if keys.include? 'fieldIndices' and keys.include? 'series'
+            x_axis = regression['fieldIndices'][0]
+            y_axis = regression['fieldIndices'][1]
+            groups = regression['fieldIndices'][2]
+            id = regression['series']['name']['id']
+            dash_style = regression['series']['dashStyle']
+            name = regression['series']['name']['group']
+            r2 = regression['series']['name']['regression']['tooltip'].split('</strong> ')[3].gsub('e', 'E')
+            type = regression['type']
+            wtf = regression['series']['name']['regression']['tooltip'].split('<br>')[1].delete('^0-9 \.\-eE').split('  ').reverse
+            params = wtf.select { |x| x != '' }
+            params[params.length - 1] = params[params.length - 1][0...params[params.length - 1].length - 1]
+            params = params.map { |x| x.gsub('e', 'E') }
+            if regression['type'] == 4 or regression['type'] == 5
+              copy = params.clone
+              params[1] = copy[2]
+              params[2] = copy[1]
+            end
+            parameters = params
+            function = get_func(regression['type'])
+            new_regression = {}
+            new_regression['type'] = type
+            new_regression['xAxis'] = x_axis
+            new_regression['yAxis'] = y_axis
+            new_regression['groups'] = groups
+            new_regression['parameters'] = parameters
+            new_regression['func'] = function
+            new_regression['id'] = id
+            new_regression['r2'] = r2
+            new_regression['name'] = name
+            new_regression['dashStyle'] = dash_style
+            scatter_regressions.push new_regression
+          else
+            scatter_regressions.push regression
+          end
+        end
+        globals['Scatter']['savedRegressions'] = scatter_regressions
+      end
+
+      if !timeline_params.nil? and timeline_params.key? 'savedRegressions'
```

Comment:
you copy pasted scatter instead of making a function?

I realize this is only a migration but like, seriously we don't have to the write the worst code ever :P